docs/first_semantic_game/02.simple_chat_npcs.rst

```diff
@@ -1,163 +1,395 @@
+holds a description of the entity. Both properties are exported for easy 
+modification in the Godot inspector.
+
+Put the following inside the ``ChatEntity`` class.
 
 .. code-block:: csharp
 
-   public partial class Player : CharacterBody2D
-   {
-       ...
+   [Export]
+   public string ChatName = "";
 
-       // Called when the node enters the scene tree for the first time.
-       public override void _Ready()
-       {
-           ...
-       }
+   [Export(PropertyHint.MultilineText)]
+   public string ChatDescr = "";
 
-       // Called once per physics tick
-       public override void _PhysicsProcess(double delta)
-       {
-           ...
-       }
-   }
+After saving the script, attach it to the ``NPC`` node in the main scene.
+Inspect the properties of the NPC, you should see ``ChatName`` and ``ChatDescr``.
+
+Next, we will add methods for handling messages: ``SendMsg`` and ``ReceiveMsg``.
 
+The ``SendMsg`` method is called when the entity wants to send a message to another
+entity. It prints the message to the console and then emits a ``MsgSent`` signal
+with the message as an argument. 
 
-Our first ChatAI class
-----------------------
+``ReceiveMsg`` is called when the entity receives a message from another entity.
+We expect this method to be overridden by a child class because the behavior of
+receiving a message will be different for different types of entities.
 
-The `ChatAI` class to be used in the demo is responsible for setting up and
-managing the conversations between NPCs, it uses OpenAI's GPT model via Semantic
-Kernel to handle the interactions. 
+.. code-block:: csharp
+
+    [Signal]
+    public delegate void MsgSentEventHandler(string msg);
 
-In this class, the following methods are used:
+    public virtual void SendMsg(string msg)
+    {
+        GD.Print(ChatName + ": " + msg);
+        CallDeferred("emit_signal", "MsgSent", msg);
+    }
+
+    public virtual void ReceiveMsg(string msg)
+    {
+        // This function will be overridden later by a child class
+    }
 
-- `GetApiKey` for retrieving the OpenAI API key from .NET secrets storage
-- A method to initialize GPT model via Semantic Kernel
-- A method to receive message and respond asynchronously
+Finally, we need to add a field to store a reference to our current conversation
+partner. This will be used later to send messages to the other entity.
+This ``_inConvoWith`` field is declared as ``internal`` so that it can be accessed
+by the main game class.
 
-This is what the base `ChatAI` class should look like:
+Now the completed ``ChatEntity`` class should look like this:
 
 .. code-block:: csharp
 
     using Godot;
-    using Microsoft.SemanticKernel;
-    using Microsoft.SemanticKernel.AI.ChatCompletion;
-    using Microsoft.Extensions.Configuration;
-    using System.Threading.Tasks;
-    
-    public class ChatAI : Node
+    using System;
+
+    public partial class ChatEntity : Area2D

```

Comment:
I had an issue where the `ChatName` and `ChatDescr` properties were not exported to the inspector. I changed this `ChatEntity` class to be a `[GlobalClass]`, rebuilt the project, and that fixed it. So I think we'll need to make sure the tutorial-user knows to add the `[GlobalClass]` attribute for `ChatEntity`.

Reply:
```suggestion

    [GlobalClass]
    public partial class ChatEntity : Area2D
```

Reply:
Also made a suggestion [here](https://github.com/RaiderSoft/semantic-godot/pull/11#discussion_r1320190964) about where we can include instructions about adding the `[GlobalClass]` attribute.